### PR TITLE
Adds install-dependencies target in Makefile

### PR DESCRIPTION
**Pull request by anvial**
## Description

Add a new Makefile target, which adds the dependency on snaps listed in the PACKAGES variable.

Here is the list of the packages:
- terraform
- golangci-lint
- go

## Environment

- Juju controller version: any

- Terraform version: any

## QA steps

```bash
make install-dependencies
```

## Links

Jira-card: [JUJU-5117](https://warthogs.atlassian.net/browse/JUJU-5117)

[JUJU-5117]: https://warthogs.atlassian.net/browse/JUJU-5117?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,3 +75,23 @@ testlxd:
...
+	@sudo snap refresh terraform --classic
+else
+	@echo "Installing terraform snap"
+	@sudo snap install terraform --classic
+endif
+
+.PHONY: install-dependencies
+install-dependencies: install-snap-dependencies
+## install-dependencies: Install all the dependencies
+	@echo "Installing dependencies"
```
Comment (hmlanigan) [inline Makefile:94]: We could add installation of `golangci-lint`

File: Makefile [hunk 2]
```diff
@@ -75,3 +75,23 @@ testlxd:
...
 ## testmicrok8s: Run unit tests against microk8s
 	TF_ACC=1 TEST_CLOUD=microk8s go test ./... -v $(TESTARGS) -timeout 120m
+
+.PHONY: install-snap-dependencies
+install-snap-dependencies:
+## install-snap-dependencies: Install snap dependencies for terraform-provider-juju
+	@echo "Installing snap dependencies"
+ifneq ($(HAS_TERRAFORM),)
+	@echo "Refreshing terraform snap"
+	@sudo snap refresh terraform --classic
```
Comment (hmlanigan) [inline Makefile:85]: Why are we doing this? The snap should refresh automatically as needed.

File: Makefile [hunk 3]
```diff
@@ -75,3 +75,23 @@ testlxd:
...
+else
+	@echo "Installing terraform snap"
+	@sudo snap install terraform --classic
```
Comment (hmlanigan) [inline Makefile:88]: In the juju Makefile we install `go` for folks also.

**Comment by hmlanigan** [comment]
@anvial Could you update the Readme to reflect these changes.

File: Makefile (hunk 2 above)
Comment (anvial) [reply Makefile:85]: I thought that without it, we would just receive the following output:

```
snap "terraform" is already installed, see 'snap help refresh'
```

Comment (anvial) [reply Makefile:85]: ok, I'll rewrite this logic... I saw something like this in our previous Makefiles

Comment (anvial) [reply Makefile:85]: I'll rewrite this login in more flexible way

File: README.md [hunk 4]
```diff
@@ -34,6 +34,12 @@ _Note:_ These features may not have functional parity with the juju cli at this
 
 1. Clone the repository
 1. Enter the repository directory
+1. Build the provider dependencies using the make `install-dependencies` target:
```
Comment (hmlanigan) [inline README.md:37]: As we're not actually building the dependencies, suggest:

"Install the provider requirements and dependencies using the make `install-dependencies` target:"

**Review by hmlanigan — APPROVED** [review]
One suggestion, but otherwise lgtm - thank you

**Pull request merged**